pkg/steps/artifacts.go

```diff
@@ -661,14 +637,13 @@ func hasMountsArtifactsVolume(pod *coreapi.Pod) bool {
 
 func gatherContainerLogsOutput(podClient PodClient, artifactDir, namespace, podName string) error {
 	var validationErrors []error
-	list, err := podClient.Pods(namespace).List(context.TODO(), meta.ListOptions{FieldSelector: fields.Set{"metadata.name": podName}.AsSelector().String()})
-	if err != nil {
+	pod := &coreapi.Pod{}
+	if err := podClient.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: namespace, Name: podName}, pod); err != nil {
```

Comment:
This is not a list. Why?

Reply:
Because a list with a `metadata.Name` selector followed by a `len(items) == 0` check is just a complicated way of doing a `Get`

Reply:
Maybe, maybe there was some reason to do it this way - I want this PR to be a client change, not a functionality change please

Reply:
It is a different way to do the exact same thing